torch/_inductor/utils.py

```diff
@@ -1558,3 +1560,16 @@ def use_scatter_fallback(
         or (reduction_type == reduce_ty and self_dtype in {torch.bool, torch.int64})
         or torch.are_deterministic_algorithms_enabled()
     )
+
+
+def tensor_is_aligned(tensor: torch.Tensor):
+    return (tensor.storage_offset() * get_dtype_size(tensor.dtype)) % ALIGNMENT == 0
+
```

Comment:
Just read your description of the second PR - we should evaluate this expression with suppressed guards. We don't want to add further compile specialization but just guide our runtime behavior.

Reply:
`with TracingContext.get().fake_mode.shape_env.suppress_guards():`